Tweeter/Controllers/TweetController.cs

```diff
@@ -35,7 +35,7 @@ public void Post([FromBody]TweetViewModel tweet)
             {
                 string user_id = User.Identity.GetUserId();
                 ApplicationUser found_app_user = Repo.Context.Users.FirstOrDefault(u => u.Id == user_id);
-                Twit found_user = Repo.Context.TweeterUsers.FirstOrDefault(twit => twit.BaseUser.UserName == found_app_user.UserName);
+                Twit found_user = Repo.GetTwitUser(found_app_user.UserName);
```

Comment:
Good - you can make this separation even more clear by making the repository's Context data member private